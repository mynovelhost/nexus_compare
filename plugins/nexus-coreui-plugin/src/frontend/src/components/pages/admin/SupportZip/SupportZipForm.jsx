/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import UIStrings from '../../../../constants/UIStrings';
import {Button, Checkbox, CheckboxGroup, NxFontAwesomeIcon} from "nexus-ui-plugin";
import {faFileArchive} from '@fortawesome/free-solid-svg-icons';

export default function SupportZipForm({params, setParams, submit, clustered, hazips}) {
  return <>
    <div dangerouslySetInnerHTML={{__html: UIStrings.SUPPORT_ZIP.DESCRIPTION}}/>
    <CheckboxGroup labelText={'Contents:'}>
      <Checkbox
        checkboxId='systemInformation'
        isChecked={params.systemInformation}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.REPORT_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='threadDump'
        isChecked={params.threadDump}
        onChange={setParams}
      >
       {UIStrings.SUPPORT_ZIP.DUMP_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='configuration'
        isChecked={params.configuration}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.CONFIGURATION_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='security'
        isChecked={params.security}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.SECURITY_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='log'
        isChecked={params.log}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.LOGFILES_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='taskLog'
        isChecked={params.taskLog}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.TASKLOGFILES_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='auditLog'
        isChecked={params.auditLog}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.AUDITLOGFILES_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='metrics'
        isChecked={params.metrics}
        onChange={setParams}
      >
        {UIStrings.SUPPORT_ZIP.METRICS_LABEL}
      </Checkbox>
      <Checkbox
        checkboxId='jmx'
        isChecked={params.jmx}
        onChange={setParams}
       >
        {UIStrings.SUPPORT_ZIP.JMX_LABEL}
       </Checkbox>
    </CheckboxGroup>
    <CheckboxGroup labelText={'Options:'}>
      <Checkbox
        checkboxId='limitFileSizes'
        isChecked={params.limitFileSizes}
        onChange={setParams}
       >
        {UIStrings.SUPPORT_ZIP.LIMITFILESIZES_LABEL}
       </Checkbox>
      <Checkbox
        checkboxId='limitZipSize'
        isChecked={params.limitZipSize}
        onChange={setParams}
       >
        {UIStrings.SUPPORT_ZIP.LIMITZIPSIZE_LABEL}
       </Checkbox>
    </CheckboxGroup>
    <Button variant='primary' onClick={submit} type='submit'>
      <NxFontAwesomeIcon icon={faFileArchive}/>
      <span>Create support ZIP</span>
    </Button>
    {clustered &&
      <Button variant='primary' onClick={hazips} type='submit'>
        <NxFontAwesomeIcon icon={faFileArchive}/>
        <span>Create support ZIP (all nodes)</span>
      </Button>
    }
  </>;
}
