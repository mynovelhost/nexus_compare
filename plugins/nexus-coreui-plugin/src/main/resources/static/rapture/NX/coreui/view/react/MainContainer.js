/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
/*global Ext, NX, REACT_COMPONENTS, react, ReactDOM */

/**
 * React Main Container
 *
 * @since 3.21
 */
Ext.define('NX.coreui.view.react.MainContainer', {
  extend: 'Ext.Component',
  alias: 'widget.nx-coreui-react-main-container',

  reactView: undefined,

  scrollable: true,

  listeners: {
    afterrender: 'initReactView',
    beforedestroy: 'destroyReactView'
  },

  initReactView: function() {
    var reactElement = react.createElement(this.reactView);
    ReactDOM.render(reactElement, this.getEl().dom);
  },

  destroyReactView: function() {
    if (this.reactView) {
      ReactDOM.unmountComponentAtNode(this.getEl().dom);
    }
  },

  refresh: function() {
    this.destroyReactView();
    this.initReactView();
  }
});
