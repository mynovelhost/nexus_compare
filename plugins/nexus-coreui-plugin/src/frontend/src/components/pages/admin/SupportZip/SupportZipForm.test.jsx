/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import React from 'react';
import { act } from 'react-dom/test-utils';
import {fireEvent, render, wait} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SupportZipForm from './SupportZipForm';

describe('SupportZipForm', function() {
  const renderView = async (params, setParams, submit, clustered, hazips) => {
    var selectors;

    await act(async () => {
      selectors = render(<SupportZipForm params={params} setParams={setParams} submit={submit} clustered={clustered} hazips={hazips} />);
    });

    return selectors;
  };

  it('renders support zip form', async function() {
    const params = {
      systemInformation: true,
      threadDump: false,
      configuration: true,
      security: false,
      log: true,
      taskLog: false,
      auditLog: true,
      metrics: false,
      jmx: true,
      limitFileSizes: false,
      limitZipSize: true
    };

    const {container} = await renderView(params);

    expect(container.querySelector('input#systemInformation')).toBeChecked();
    expect(container.querySelector('input#threadDump')).not.toBeChecked();
    expect(container.querySelector('input#configuration')).toBeChecked();
    expect(container.querySelector('input#security')).not.toBeChecked();
    expect(container.querySelector('input#log')).toBeChecked();
    expect(container.querySelector('input#taskLog')).not.toBeChecked();
    expect(container.querySelector('input#auditLog')).toBeChecked();
    expect(container.querySelector('input#metrics')).not.toBeChecked();
    expect(container.querySelector('input#jmx')).toBeChecked();
    expect(container.querySelector('input#limitFileSizes')).not.toBeChecked();
    expect(container.querySelector('input#limitZipSize')).toBeChecked();  

    expect(container).toMatchSnapshot();
  });

  it('updates the parameters when clicked', async function() {
    const params = {
      systemInformation: true,
      threadDump: false,
      configuration: true,
      security: false,
      log: true,
      taskLog: false,
      auditLog: true,
      metrics: false,
      jmx: true,
      limitFileSizes: false,
      limitZipSize: true
    };
    const setParams = jest.fn((event) => event.persist());
    const {container} = await renderView(params, setParams);

    Object.entries(params).forEach(([name, value]) => {
      fireEvent.click(container.querySelector(`input#${name}`));
      expect(setParams.mock.calls.slice(-1)[0][0].target.id).toBe(name);
    });
    expect(setParams).toHaveBeenCalledTimes(Object.entries(params).length);
  });

  it('renders button for all nodes only when clustered', async function() {
    const params = {
      systemInformation: true,
      threadDump: false,
      configuration: true,
      security: false,
      log: true,
      taskLog: false,
      auditLog: true,
      metrics: false,
      jmx: true,
      limitFileSizes: false,
      limitZipSize: true
    };

    const nonHa = (await renderView(params, undefined, undefined, false, undefined)).container;
    expect(nonHa.querySelectorAll('button')).toHaveLength(1);

    const ha = (await renderView(params, undefined, undefined, true, undefined)).container;
    expect(ha.querySelectorAll('button')).toHaveLength(2);
  });
});
