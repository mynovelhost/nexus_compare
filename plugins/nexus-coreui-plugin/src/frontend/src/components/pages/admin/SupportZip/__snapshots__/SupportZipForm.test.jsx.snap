// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupportZipForm renders support zip form 1`] = `
<div>
  <div>
    <p>
      No information will be sent to Sonatype when creating the support ZIP file.
    </p>
    <p>
      Support ZIP creation may take a few minutes to complete.
    </p>
  </div>
  <div
    class="nxrm-checkbox-group"
  >
    <label
      class="nxrm-checkbox-group-label"
    >
      Contents:
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="systemInformation"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        System information report
      </span>
    </label>
    <label
      class="nx-checkbox tm-unchecked"
    >
      <input
        class="nx-checkbox__input"
        id="threadDump"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
         
      </span>
      <span
        class="nx-checkbox__content"
      >
        JVM thread-dump
      </span>
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="configuration"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        Configuration files
      </span>
    </label>
    <label
      class="nx-checkbox tm-unchecked"
    >
      <input
        class="nx-checkbox__input"
        id="security"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
         
      </span>
      <span
        class="nx-checkbox__content"
      >
        Security configuration files
      </span>
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="log"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        Log files
      </span>
    </label>
    <label
      class="nx-checkbox tm-unchecked"
    >
      <input
        class="nx-checkbox__input"
        id="taskLog"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
         
      </span>
      <span
        class="nx-checkbox__content"
      >
        Task log files
      </span>
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="auditLog"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        Audit log files
      </span>
    </label>
    <label
      class="nx-checkbox tm-unchecked"
    >
      <input
        class="nx-checkbox__input"
        id="metrics"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
         
      </span>
      <span
        class="nx-checkbox__content"
      >
        System and component metrics
      </span>
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="jmx"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        JMX information
      </span>
    </label>
  </div>
  <div
    class="nxrm-checkbox-group"
  >
    <label
      class="nxrm-checkbox-group-label"
    >
      Options:
    </label>
    <label
      class="nx-checkbox tm-unchecked"
    >
      <input
        class="nx-checkbox__input"
        id="limitFileSizes"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
         
      </span>
      <span
        class="nx-checkbox__content"
      >
        Limit files in the ZIP archive to 30 MB apiece
      </span>
    </label>
    <label
      class="nx-checkbox tm-checked"
    >
      <input
        checked=""
        class="nx-checkbox__input"
        id="limitZipSize"
        readonly=""
        type="checkbox"
      />
      <span
        class="nx-checkbox__box"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-check fa-w-16 "
          data-icon="check"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
            fill="currentColor"
          />
        </svg>
      </span>
      <span
        class="nx-checkbox__content"
      >
        Limit the ZIP archive to 20 MB
      </span>
    </label>
  </div>
  <button
    class="nx-btn nx-btn--primary"
    type="submit"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-file-archive fa-w-12 nx-icon"
      data-icon="file-archive"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M377 105L279.1 7c-4.5-4.5-10.6-7-17-7H256v128h128v-6.1c0-6.3-2.5-12.4-7-16.9zM128.4 336c-17.9 0-32.4 12.1-32.4 27 0 15 14.6 27 32.5 27s32.4-12.1 32.4-27-14.6-27-32.5-27zM224 136V0h-63.6v32h-32V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zM95.9 32h32v32h-32zm32.3 384c-33.2 0-58-30.4-51.4-62.9L96.4 256v-32h32v-32h-32v-32h32v-32h-32V96h32V64h32v32h-32v32h32v32h-32v32h32v32h-32v32h22.1c5.7 0 10.7 4.1 11.8 9.7l17.3 87.7c6.4 32.4-18.4 62.6-51.4 62.6z"
        fill="currentColor"
      />
    </svg>
    <span>
      Create support ZIP
    </span>
  </button>
</div>
`;
